app/assets/javascripts/media_player_wrapper/avalon_player_new.es6

```diff
@@ -119,6 +145,63 @@ class MEJSPlayer {
   }
 
   /**
+   * Create an element representing highlighted segment area in MEJS's time rail
+   * and add it to the DOM
+   * @function createTimeHighlightEl
+   * @return {void}
+   */
+  createTimeHighlightEl () {
+    let highlightSpanEl = document.getElementById('content').querySelector('.mejs-highlight-clip')
+
+    // Create the highlight DOM element if doesn't exist
+    if (!highlightSpanEl) {
+      highlightSpanEl = document.createElement('span')
+      highlightSpanEl.classList.add('mejs-highlight-clip')
+      $('.mejs__time-total')[0].appendChild(highlightSpanEl)
+      this.highlightSpanEl = highlightSpanEl
+    }
+  }
+
+  /**
+   * Create a CSS 'style' string to plug into an inline style attribute
+   * @function createTimeRailStyles
+   * @return {string} Ex. left: 10%; width: 50%;
+   */
+  createTimeRailStyles (activeSegmentId) {
+    const duration = this.currentStreamInfo.duration
+    const segment = this.segmentsMap[activeSegmentId]
+    let t = []
+
+    // No active segment, remove highlight style
+    if (!activeSegmentId) {
+      return 'left: 0%; width: 0%;'
+    }
+
+    // Use the active segment fragment,
+    try {
+      t = [parseFloat(segment.fragmentbegin), parseFloat(segment.fragmentend)]
+    } catch(e) {
+      t = this.currentStreamInfo.t.slice(0)
+    }
+
+    // Ensure t range array has valid values
+    t[0] = (isNaN(parseFloat(t[0]))) ? 0 : t[0]
+    t[1] = (t.length < 2 || isNaN(parseFloat(t[1]))) ? 100 : t[1]
```

Comment:
I think the fallback value here should be duration, not 100.

Reply:
Good call. Think I got this mixed up with "100%".